### PR TITLE
Display "Share" alt text for share action icon

### LINKED ISSUE #19089: Accessibility of 'share button', alternative text is not provided for the image that is the only thing within a link.
### Steps to reproduce

1. Login ownCloud.

### Expected behaviour
An alternative text should be provided for 'share button' due to it is a image that is the only thing within a link.

### Actual behaviour

An alternative text is not provided for 'share button' image.

### Server configuration
**Operating system**:
Ubuntu 14.04

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
master 
{"installed":true,"maintenance":false,"version":"8.2.0.4","versionstring":"8.2 pre alpha","edition":""}

**Updated from an older ownCloud or fresh install:**
Fresh

**List of activated apps:**

**The content of config/config.php:**

```

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** 
No

**Logs**
```
```
### Client configuration

**browser**
 Firefox.

### PR DESCRIPTION
**Pull request by PVince81**
## Description

Display "Share" alt text on share action icon for accessibility.
## Related Issue

Fixes https://github.com/owncloud/core/issues/19089
## Motivation and Context

Accessibility
## How Has This Been Tested?

Needs DOM inspector:
- [x] TEST: text "Share" appears in "hidden-visually" block when a file is not shared
- [x] TEST: text "Shared" appears in a regular label block (no "hidden-visually") when a file is shared
- [x] TEST: switching between shared and not shared properly updates the label state, switching between "hidden-visually" when no text and the regular span
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Please review @davitol @VicDeo

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by davitol** [comment]
tested and fixes https://github.com/owncloud/core/issues/19089 :+1:

File: core/js/share.js [hunk 1]
```diff
@@ -306,7 +306,7 @@ OC.Share = _.extend(OC.Share || {}, {
 			}
 		}
 		else {
-			action.html('<span></span>').prepend(icon);
+			action.html('<span class="hidden-visually">' + t('core', 'Shared') + '</span>').prepend(icon);
```
Comment (jancborchardt) [inline core/js/share.js]: @PVince81 This leads to a non-shared file having a span saying »Shared« next to it.

Also, the case I mentioned in https://github.com/owncloud/core/issues/19089#issuecomment-147273450 isn’t fixed:

> In the case of another person having it shared with you, the alt text of the icon should say »Shared by«.

There still only the icon is shown.

Comment (PVince81) [inline core/js/share.js]: Ah damn, it was supposed to say "Share", not "Shared" to restore it to its original state

**Pull request merged**